### PR TITLE
Migration to babel 7 and cleaned up eslint

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Migration to babel 7 and cleaned up eslint

#### Where should the reviewer start?

everywhere

#### What testing has been done on this PR?

storybook, unit tests

#### How should this be manually tested?

storybook

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/2338

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "presets": [
-    "./tools/grommet-babel-preset-es2015",
```
Comment (alansouzati) [inline .babelrc]: removing this is breaking the es6 distribution.

we run this

`cross-env BABEL_ENV=es6 babel ./src/js/ --ignore '__tests__' --out-dir ./dist/es6 --copy-files`

and `./tools/grommet-babel-preset-es2015` is the one responsible for adding `es6` as the babel environment.

You can test this by opening the `dist/es6/components/Accordion/Accordion.js`.

This is important for tree shaking

File: src/js/components/Accordion/Accordion.js [hunk 2]
```diff
@@ -82,7 +78,7 @@ class Accordion extends Component {
       <Box role='tablist' {...rest} overflow='auto'>
         {Children.toArray(children).map((panel, index) => (
           <AccordionContext.Provider
-            key={`accordion-panel_${index}`}
+            key={`accordion-panel_${index + 0}`}
```
Comment (alansouzati) [inline src/js/components/Accordion/Accordion.js]: why is this needed?

File: tools/generate-readme-ts.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import del from 'del';
+import del from 'del'; // eslint-disable-line import/no-extraneous-dependencies
```
Comment (alansouzati) [inline tools/generate-readme-ts.js]: why are we disabling this? isn't `del` in the package.json ?

File: storybook/config.js [hunk 4]
```diff
@@ -1,15 +1,18 @@
-import { configure } from '@storybook/react';
-import { setOptions } from '@storybook/addon-options';
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (alansouzati) [inline storybook/config.js]: why is this here?

File: storybook/addons.js [hunk 5]
```diff
@@ -1,2 +1,3 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (alansouzati) [inline storybook/addons.js]: same..

File: src/js/components/Anchor/anchor.stories.js [hunk 6]
```diff
@@ -19,6 +19,8 @@ storiesOf('Anchor', module)
   ))
   .add('With Text', () => (
     <Grommet theme={grommet}>
-      This is a <Anchor label='link' href='#' /> with text.
```
Comment (alansouzati) [inline src/js/components/Anchor/anchor.stories.js]: this needs to be a single otherwise it will show like this:

screen shot 2018-10-12 at 12 38 23 am

File: src/js/components/DataTable/datatable.stories.js [hunk 7]
```diff
@@ -58,51 +60,55 @@ for (let i = 0; i < 40; i += 1) {
...
 const DATA = [
-  { name: 'Alan', location: 'Los Gatos', date: '2018-06-11', percent: 20, paid: 2345 },
-  { name: 'Bryan', location: 'Fort Collins', date: '2018-06-10', percent: 30, paid: 1234 },
-  { name: 'Chris', location: 'Palo Alto', date: '2018-06-09', percent: 40, paid: 2345 },
-  { name: 'Eric', location: 'Palo Alto', date: '2018-06-11', percent: 80, paid: 3456 },
-  { name: 'Doug', location: 'Fort Collins', date: '2018-06-10', percent: 60, paid: 1234 },
-  { name: 'Jet', location: 'Palo Alto', date: '2018-06-09', percent: 40, paid: 3456 },
-  { name: 'Michael', location: 'Boise', date: '2018-06-11', percent: 50, paid: 1234 },
-  { name: 'Tracy', location: 'San Francisco', date: '2018-06-10', percent: 10, paid: 2345 },
+  {
```
Comment (alansouzati) [inline src/js/components/DataTable/datatable.stories.js]: for some reason Tunable and Served stories are broken. the first two columns have background white. this seems like a regression.

File: src/js/components/Accordion/Accordion.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Accordion/Accordion.js]: It avoids eslint complaints about using index as a key.
Would you prefer disabling the eslint rule? I actually wonder if an inline /* eslint-disable */ would be even better, to make it clearer. I didn't want to disable the eslint rule globally because sometimes there is a way to use a non-index key, and we should do that wherever possible.

File: tools/generate-readme-ts.js (hunk 3 above)
Comment (ericsoderberghp) [reply tools/generate-readme-ts.js]: eslint was complaining.

File: storybook/config.js (hunk 4 above)
Comment (ericsoderberghp) [reply storybook/config.js]: same as for tools/readme-generate-ts.js

File: src/js/components/DataTable/datatable.stories.js (hunk 7 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/datatable.stories.js]: Great catch! Fixed.
